components/create-bounty/create-bounty-token-amount.tsx

```diff
@@ -0,0 +1,186 @@
+import { useEffect, useState } from "react";
+import { NumberFormatValues } from "react-number-format";
+
+import BigNumber from "bignumber.js";
+import { useTranslation } from "next-i18next";
+import getConfig from "next/config";
+
+import DoubleArrowRight from "assets/icons/double-arrow-right";
+
+import { useAppState } from "../../contexts/app-state";
+import { getCoinPrice } from "../../services/coingecko";
+import Button from "../button";
+import InputNumber from "../input-number";
+import TokensDropdown from "../tokens-dropdown";
+
+export default function CreateBountyTokenAmount({
+  currentToken,
+  setCurrentToken,
+  addToken,
+  canAddCustomToken,
+  defaultToken = null,
+  userAddress,
+  customTokens,
+  labelSelect,
+  tokenBalance,
+  issueAmount,
+  setIssueAmount,
+  review = false,
+  needValueValidation,
+  decimals = 18,
+  isFunding = false,
+}) {
+  const { t } = useTranslation("bounty");
+  const { state } = useAppState();
+  const { publicRuntimeConfig } = getConfig();
+  const [inputError, setInputError] = useState("");
+  const [convertedAmount, setConvertedAmount] = useState(0);
+
+  function handleIssueAmountOnValueChange(values: NumberFormatValues) {
+    if (
+      needValueValidation &&
+      +values.floatValue > +currentToken?.currentValue
+    ) {
+      setIssueAmount({ formattedValue: "" });
+      setInputError(t("bounty:errors.exceeds-allowance"));
+    } else if (values.floatValue < 0) {
+      setIssueAmount({ formattedValue: "" });
+    } else if (
+      values.floatValue !== 0 &&
+      BigNumber(values.floatValue).isLessThan(BigNumber(state.Settings?.minBountyValue))
+    ) {
+      setInputError(t("bounty:errors.exceeds-minimum-amount", {
+          amount: state.Settings?.minBountyValue,
+      }));
+    } else {
+      setIssueAmount(values);
+      if (inputError) setInputError("");
+    }
+  }
+
+  function handleIssueAmountBlurChange() {
+    if (needValueValidation && tokenBalance?.lt(issueAmount.floatValue)) {
+      setIssueAmount({ formattedValue: tokenBalance.toFixed() });
+    }
+  }
+
+  function updateConversion() {
+    if (!currentToken?.symbol || !publicRuntimeConfig?.enableCoinGecko) return;
+
+    getCoinPrice(currentToken?.symbol,
+                 state?.Settings?.currency?.defaultFiat).then((price) => {
+                   setConvertedAmount(issueAmount.value * price);
+                 });
+  }
+
+  function selectTokens() {
+    return (
+      <TokensDropdown
+        token={currentToken}
+        label={labelSelect}
+        tokens={customTokens}
+        userAddress={userAddress}
+        canAddToken={canAddCustomToken}
+        addToken={addToken}
+        setToken={setCurrentToken}
+        disabled={review}
+        defaultToken={defaultToken}
+        showCurrencyValue={false}
+        needsBalance={isFunding}
+        noLabel
+      />
+    );
+  }
+
+  function inputNumber() {
```

Comment:
inputNumber is not re-used, we gain nothing but complication by having it as a function